packages/web-config-server/src/apiV1/dataBuilders/generic/analytics/analyticsYearOnYear.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Tupaia Config Server
+ * Copyright (c) 2019 Beyond Essential Systems Pty Ltd
+ */
+
+import { formatLayeredDataElementCode, layerYearOnYear } from '../../../utils/layerYearOnYear';
+import { AnalyticsPerPeriodBuilder } from './analyticsPerPeriod';
+import { momentToPeriod, periodToMoment, reduceToDictionary } from '@tupaia/utils';
+import moment from 'moment';
+
+class AnalyticsYearOnYearBuilder extends AnalyticsPerPeriodBuilder {
+  async fetchAnalytics(
+    dataElementCodes,
+    additionalQueryConfig,
+    aggregationType = this.aggregationType,
+    aggregationConfig = this.config.aggregationConfig ?? {},
+  ) {
+    const originalStartDate = this.query.startDate;
+    const originalEndDate = this.query.endDate;
+
+    this.query.endDate = this.getDataEndDate().format('YYYY-MM-DD');
+    this.query.startDate = this.getDataStartDate().format('YYYY-MM-DD');
+
+    const response = await super.fetchAnalytics(
+      dataElementCodes,
+      additionalQueryConfig,
+      aggregationType,
+      aggregationConfig,
+    );
+
+    const layeredAnalytics = layerYearOnYear(response.results);
+    response.results = this.filterLayeredAnalytics(
+      originalStartDate,
+      originalEndDate,
+      layeredAnalytics,
+    );
+
+    this.query.startDate = originalStartDate;
+    this.query.endDate = originalEndDate;
+
+    return response;
+  }
+
+  filterLayeredAnalytics(originalStartDate, originalEndDate, layeredAnalytics) {
+    let filteredAnalytics = layeredAnalytics.filter(layeredAnalytic => {
+      let analytic = periodToMoment(layeredAnalytic.period);
+      analytic = analytic.format('YYYY-MM-DD');
```

Comment:
```suggestion
      const formattedAnalyticPeriod = periodToMoment(layeredAnalytic.period).format('YYYY-MM-DD');
```
Always nice to give variables a clear name representing what they are 👍